src/background/storage.js

```diff
@@ -35,33 +36,30 @@ class Storage {
     try {
       let storagePersistNeeded = false;
       this.local = await browser.storage.local.get();
+      this.preferences = await browser.storage.sync.get();
       if (!Object.keys(this.local).length) {
         this.local = {
           tempContainerCounter: 0,
           tempContainers: {},
-          tabContainerMap: {},
-          preferences: this.preferencesDefault
+          tabContainerMap: {}
         };
         debug('storage empty, setting defaults');
         storagePersistNeeded = true;
       } else {
         debug('storage loaded', this.local);
       }
+
       // set preferences defaults if not present
-      if (!this.local.preferences) {
+      if (!Object.keys(this.preferences).length) {
```

Comment:
This will always be true for a fresh installation if I'm not mistaken. Since this is the migration path it should additionally check whether `this.local.preferences` has keys to migrate.

Reply:
Actually that's what is done the next line, when we create this.preferences we make sure to override default preferences with the local ones.

Reply:
If you install the Add-on the first time after this change is published, `this.local.preferences` will be `undefined` and the migration happening is not needed. It will only happen once; but still - it should be a migration-path, not something that runs on every first time install.

Reply:
Ok, I see now - it actually does two things. For one it sets the default on first time install - which is wanted behavior. And additionally it migrates `this.local.preferences` if available. Could you please check explicitly for `this.local.preferences` inside this if and only do the migration if exists? Thanks.

Reply:
I can but it doesn't cost anything to do it if it doesn't exists :)